src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -342,11 +349,11 @@ public List<SourceFile> listSourceFiles(Iterable<NamedStyles> styles,
         sourceFiles.addAll(ListUtils.map(maybeAutodetectStyles(javaParser.parse(mainJavaSources, baseDir, ctx), styles),
                 addProvenance(baseDir, projectProvenance, generatedSourcePaths)));
 
-        ResourceParser rp = new ResourceParser(logger, exclusions, sizeThresholdMb);
+        ResourceParser rp = new ResourceParser(baseDir, logger, exclusions, sizeThresholdMb, skipOtherMavenProjects());
 
         // Any resources parsed from "main/resources" should also have the main source set added to them.
         sourceFiles.addAll(ListUtils.map(
-                rp.parse(baseDir, mavenProject.getBasedir().toPath().resolve("src/main/resources"), alreadyParsed),
+                rp.parse(mavenProject.getBasedir().toPath().resolve("src/main/resources"), alreadyParsed),
```

Comment:
I am a little concerned here because the `baseDir` is the root of the maven project and all paths are made relative to it....this helps when we are checking for exclusions...

To clarify here the instance variable `baseDir` should be the root directory for the entire maven project and `mavenProject.getBaseDir()` will be the directory of the current project.

I could be wrong here and maybe you are accounting for this below....just an observation.

Reply:
The `baseDir` is passed into the ResourceParser constructor on line 352.

Prior to the change, PathMatchers were generated using `baseDir.getFileSystem().getPathMatcher(..)` per path that is traversed. By moving the `baseDir` into the constructor, we don't need to pass the implied path into each method call. And the PathMatchers may be generated during instantiation of ResourceParser rather than doing the work repeatedly.